src/aatoolbox/datasources/iri/iri_seasonal_forecast.py

```diff
@@ -0,0 +1,269 @@
+
+        Parameters
+        ----------
+        iri_auth: str
+            iri key for authentication. An account is
+            needed to get this key config. For an account this key might
+            be changed over time, so might need to update it regularly
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+        """
+        output_filepath = self._get_raw_path()
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
+
+        url = self._get_url()
+        logger.info("Downloading IRI NetCDF file.")
+        return self._download(
+            filepath=output_filepath,
+            url=url,
+            iri_auth=iri_auth,
+            clobber=clobber,
+        )
+
+    @staticmethod
+    @check_file_existence
+    def _download(filepath, url, iri_auth, clobber):
+        response = requests.get(
+            url,
+            # have to authenticate by using a cookie
+            cookies={"__dlauth_id": iri_auth},
+        )
+        with open(filepath, "wb") as out_file:
+            out_file.write(response.content)
+        return filepath
+
+    def process(self, clobber: bool = False):
+        """
+        Process the IRI forecast.
+
+        Parameters
+        ----------
+        clobber : bool, default = False
+            If True, overwrites existing processed files
+
+        """
+        ds = xr.load_dataset(
+            self._get_raw_path(),
+            decode_times=False,
+            drop_variables="C",
+        )
+        processed_file_path = self._get_processed_path()
+        processed_file_path.parent.mkdir(parents=True, exist_ok=True)
+        return self._process(
+            filepath=processed_file_path, ds=ds, clobber=clobber
+        )
+
+    @staticmethod
+    @check_file_existence
+    def _process(filepath: str, ds, clobber: bool):
+        # fix dates
+        ds.aat.set_time_dim(t_dim="F", inplace=True)
+        ds.aat.correct_calendar(inplace=True)
+        ds = xr.decode_cf(ds)
+
+        # IRI downloads in the order you give the coordinates
+        # so make sure to invert them
+        # IRI accepts -180 to 180 longitudes and 0 to 360
+        # but automatically converts them to -180 to 180
+        # so we don't need to do that
+        # TODO: can be removed once we have a check in the
+        # geoboundingbox class for south<north
+        # TODO: for some reason the `inplace` is not working
+        # re-add when we fixed that
+        ds = ds.aat.invert_coordinates()
+        ds.to_netcdf(filepath)
+        return filepath
+
+    def load(self):
+        """Load the IRI forecast."""
+        ds = xr.load_dataset(self._get_processed_path())
+        return ds.rio.write_crs("EPSG:4326", inplace=True)
+
+    def _get_file_name(self):
+        file_name = f"{self._iso3}_iri_forecast_seasonal_precipitation_tercile"
+        if self._forecast_type == "dominant":
+            file_name += "_dominant"
```

Comment:
You should use the variable `_forecast_type` as the string, e.g. `file_name += f"_self._forecast_type"`